Dockerfile

```diff
@@ -104,73 +87,79 @@ RUN apt-get -y install \
+    PYTHON_VERSION_36=3.6.8 \
+    PYTHON_VERSION_37=3.7.2 \
+    CONDA_VERSION=4.5.12
+LABEL python.version_27=$PYTHON_VERSION_27 \
+      python.version_36=$PYTHON_VERSION_36 \
+      python.version_37=$PYTHON_VERSION_37 \
+      conda.version=$CONDA_VERSION
+
 # Install Conda
-RUN curl -O https://repo.continuum.io/miniconda/Miniconda2-${CONDA_VERSION}-Linux-x86_64.sh
-RUN bash Miniconda2-${CONDA_VERSION}-Linux-x86_64.sh -b -p /home/docs/.conda/
-ENV PATH $PATH:/home/docs/.conda/bin
-RUN rm -f Miniconda2-${CONDA_VERSION}-Linux-x86_64.sh
+RUN curl -O https://repo.continuum.io/miniconda/Miniconda2-${CONDA_VERSION}-Linux-x86_64.sh \
+ && bash Miniconda2-${CONDA_VERSION}-Linux-x86_64.sh -b -p /home/docs/.conda/ \
+ && rm -f Miniconda2-${CONDA_VERSION}-Linux-x86_64.sh
+ENV PATH=$PATH:/home/docs/.conda/bin
 
 # Install pyenv
-RUN wget https://github.com/pyenv/pyenv/archive/master.zip
-RUN unzip master.zip && \
-    rm -f master.zip && \
-    mv pyenv-master ~docs/.pyenv
-ENV PYENV_ROOT /home/docs/.pyenv
-ENV PATH /home/docs/.pyenv/shims:$PATH:/home/docs/.pyenv/bin
+RUN curl -L https://github.com/pyenv/pyenv/archive/master.tar.gz \
+  | tar xz \
+ && mv pyenv-master .pyenv
+ENV PYENV_ROOT=/home/docs/.pyenv \
+    PATH=/home/docs/.pyenv/shims:$PATH:/home/docs/.pyenv/bin
 
 # Install supported Python versions
-RUN pyenv install $PYTHON_VERSION_27 && \
-    pyenv install $PYTHON_VERSION_37 && \
-    pyenv install $PYTHON_VERSION_36 && \
-    pyenv global \
-        $PYTHON_VERSION_27 \
-        $PYTHON_VERSION_37 \
-        $PYTHON_VERSION_36
-
-WORKDIR /tmp
-
-RUN pyenv local $PYTHON_VERSION_27 && \
-    pyenv exec pip install --no-cache-dir -U pip && \
-    pyenv exec pip install --no-cache-dir --only-binary numpy,scipy numpy scipy && \
-    pyenv exec pip install --no-cache-dir pandas matplotlib virtualenv
-
-RUN pyenv local $PYTHON_VERSION_37 && \
-    pyenv exec pip install --no-cache-dir -U pip && \
-    pyenv exec pip install --no-cache-dir --only-binary numpy,scipy numpy scipy && \
-    pyenv exec pip install --no-cache-dir pandas matplotlib virtualenv
-
-RUN pyenv local $PYTHON_VERSION_36 && \
-    pyenv exec pip install --no-cache-dir -U pip && \
-    pyenv exec pip install --no-cache-dir --only-binary numpy,scipy numpy scipy && \
-    pyenv exec pip install --no-cache-dir pandas matplotlib virtualenv
+RUN set -x \
+ && pyenv install "${PYTHON_VERSION_27}" \
+ && pyenv install "${PYTHON_VERSION_37}" \
+ && pyenv install "${PYTHON_VERSION_36}" \
+ && pyenv global \
+      "${PYTHON_VERSION_27}" \
+      "${PYTHON_VERSION_37}" \
+      "${PYTHON_VERSION_36}" \
+ && true
+
+RUN set -x \
+ && cd /tmp \
+ && for env in "${PYTHON_VERSION_27}" "${PYTHON_VERSION_37}" "${PYTHON_VERSION_36}"; \
+    do   pyenv local "${env}" \
+      && pyenv exec pip install --no-cache-dir -U pip \
+      && pyenv exec pip install --no-cache-dir --only-binary numpy,scipy \
+           matplotlib \
+           numpy \
+           pandas \
+           scipy \
```

Comment:
BTW, I think we should remove these packages from the default image, they add no value at all:

- no pinned versions, so user can't trust on them
- now, they come already compiled, so no time saved for builds
- most users that need these packages uses conda anyway

More of this on #90

Reply:
I'll do whatever you say here.

But actually it wasn't my focus (yet) to change the docker image by its interface. I just made stylistic changes , which shouldn't change the actual interface.

Can we delay the "remove superfluous packages" until a new PR please? There are many `-dev` packages in the final image, which are - according to my understanding - not necessary.

Reply:
Yes, it has not to be done in this PR.